### PR TITLE
MySQL binding: allow passing parameters for queries

### LINKED ISSUE #2973: MySQL binding: add support for parameters in queries
## Describe the feature

The MySQL binding doesn't support parametrized queries. Using parameters is extremely important to prevent SQL injection attacks.

## Release Note

RELEASE NOTE: **ADD** Support for parametrized queries to MySQL binding

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #2973 

Adds support for query parameters in the MySQL binding. This allows executing parametrized queries (such as `SELECT * FROM table WHERE id = ?`) which are extremely important to prevent SQL injections.

PS: this is an alpha component and doesn't have certification tests. The feature is being validated in the integration tests (which don't run in the CI but pass locally)

### CODE REVIEW

File: bindings/mysql/mysql.go [hunk 1]
```diff
@@ -87,21 +90,22 @@ type mysqlMetadata struct {
 
 	// ConnMaxIdleTime is the maximum amount of time a connection may be idle.
 	ConnMaxIdleTime time.Duration `mapstructure:"connMaxIdleTime"`
-
-	// MaxRetries is the maximum number of retries for a query.
-	MaxRetries int `mapstructure:"maxRetries"`
```
Comment (ItalyPaleAle) [inline bindings/mysql/mysql.go:92]: This metadata property was completely unused and was removed

File: bindings/mysql/mysql.go [hunk 2]
```diff
@@ -254,17 +282,25 @@ func initDB(url, pemPath string) (*sql.DB, error) {
...
+		})
 		if err != nil {
 			return nil, fmt.Errorf("error register TLS config: %w", err)
 		}
 	}
 
-	db, err := sql.Open("mysql", url)
+	// Required to correctly parse time columns
+	// See: https://stackoverflow.com/a/45040724
+	conf.ParseTime = true
```
Comment (ItalyPaleAle) [inline bindings/mysql/mysql.go:296]: Integration tests were failing without this

File: bindings/mysql/mysql.go [hunk 3]
```diff
@@ -112,23 +116,29 @@ func (m *Mysql) Init(ctx context.Context, md bindings.Metadata) error {
...
+	m.db, err = initDB(meta.URL, meta.PemPath)
 	if err != nil {
 		return err
 	}
 
-	db.SetMaxIdleConns(meta.MaxIdleConns)
-	db.SetMaxOpenConns(meta.MaxOpenConns)
-	db.SetConnMaxIdleTime(meta.ConnMaxIdleTime)
-	db.SetConnMaxLifetime(meta.ConnMaxLifetime)
+	if meta.MaxIdleConns > 0 {
```
Comment (berndverst) [inline bindings/mysql/mysql.go:124]: possibly a behavior change here right? Not sure what setting the value to 0 (for unused metadata) would have done in all of those cases.

Comment (berndverst) [reply bindings/mysql/mysql.go:124]: The behavior now is correct - but unclear if something is changing for users or not.

**Review by berndverst — APPROVED** [review]

File: bindings/mysql/mysql.go (hunk 3 above)
Comment (ItalyPaleAle) [reply bindings/mysql/mysql.go:124]: From the APIs:

image

image

image

**Pull request merged**